tests/digraph/test_dist_matrix.py

```diff
@@ -98,4 +98,48 @@ def test_digraph_distance_matrix_parallel_as_undirected(self):
                 [1.0, 2.0, 3.0, 3.0, 2.0, 1.0, 0.0],
             ]
         )
-        self.assertTrue(numpy.array_equal(dist, expected))
+        self.assertTrue(np.array_equal(dist, expected))
+
+    def test_digraph_distance_matrix_non_zero_null(self):
+        graph = retworkx.PyDiGraph()
+        graph.add_nodes_from(list(range(7)))
+        graph.add_edges_from_no_data(
+            [(0, 1), (0, 6), (1, 2), (2, 3), (3, 4), (4, 5), (5, 6)]
+        )
+        dist = retworkx.distance_matrix(
+            graph, as_undirected=True, null_value=np.inf
+        )
+        expected = np.array(
+            [
+                [np.inf, 1.0, 2.0, 3.0, 3.0, 2.0, 1.0],
```

Comment:
Still feels like the distance from a node to itself should be `0.0`, especially as undirected -- at worst this would be 7

Reply:
That's a good point. When I was writing the PR I was annoyed at first that there were no `inf` in the output, but you're totally right we should be setting this to 0.0 not `inf` and only use `inf` for truly disconnected nodes in the graph.

Reply:
I updated the logic in: https://github.com/Qiskit/retworkx/pull/410/commits/20594eca9fba2faec6d8f73e8f1fe48a68ebc1ee